packages/cfd/src/features/Containers/mt5-trade-modal.tsx

```diff
@@ -18,76 +33,228 @@ type TMT5TradeModalProps = {
         arg5: string | undefined
     ) => void;
     toggleModal: () => void;
-    is_demo: string;
 };
 
-const MT5TradeModal = observer(
-    ({ is_eu_user, is_open, onPasswordManager, toggleModal, is_demo }: TMT5TradeModalProps) => {
-        const { traders_hub, common, ui } = useStore();
-
-        const { show_eu_related_content } = traders_hub;
-        const { platform } = common;
-        const { is_mobile } = ui;
+const getTitle = (market_type: string, show_eu_related_content: boolean) => {
+    if (show_eu_related_content) localize('MT5 CFDs');
+    return market_type;
+};
 
-        const { mt5_trade_account, dxtrade_tokens, derivez_tokens, ctrader_tokens } = useCfdStore();
+const DMT5TradeModal = ({
```

Comment:
@hirad-deriv what is the plan on having a proper shared trade modal, if im not mistaken we've planned for it several months back.

Reply:
We separated the modal into 2 modals, one for other cfds and one for MT5 as discussed previously.